src/core/shape_inference/include/dimension_util.hpp

```diff
@@ -216,6 +216,25 @@ auto floor_div(const TDim& dim, const typename TDim::value_type divisor) -> TDim
     }
 }
 
+/**
+ * @brief Check if dimension has valid value.
+ *
+ * @tparam TDim     Dimension type.
+ * @param d  Dimension for check.
+ * @return true if dimension has valid value.
+ */
+template <class TDim,
+          typename std::enable_if<!std::is_same<Dimension, typename std::decay<TDim>::type>::value>::type* = nullptr>
+bool is_valid(TDim&& d) {
+    return d != typename std::decay<TDim>::type{};
```

Comment:
I think will replace it the name should be better. The `is_valid` was used because of validation context.

The `is_valid` or `is_empty` it doesn't mean that dimension has `0` value. As for `ov::Dimensino` these two states are different but for static dimension as it is simpler these two states unify to same internal value.

But in general the check `is_empty` is different then `dimension == 0`.